config/config.go

```diff
@@ -29,6 +31,12 @@ type Config struct {
 	PlannedMaintenanceEnd      string       `env:"PLANNED_MAINTENANCE_END_TIME"   flag:"planned-maintenance-end-time"    flagDesc:"The time of the day at which Planned E5 maintenance ends"`
 }
 
+// PenaltyDetailsMap defines the struct to hold the penalty details
+type PenaltyDetailsMap struct {
+	Description string                       `yaml:"name"`
+	Details     map[string]map[string]string `yaml:"details"`
```

Comment:
Adding the comment to track it, but open question on if we can find a more elegant solution to this instead of nested maps. Wonder if there's a means of doing this with objects or different file type (json?) perhaps, or even having all the fields separated and not in map. We can address it further on CS/ROE updates anyways.

Reply:
I think with JSON we would be in the same boat. The reason I went for YAMl was that there is already a yaml file used for the 'penalty_types' and wanted to keep it consistent.I had tried to use a 'Penalty' object as the value in map but ran into issues when unmarshalling.

Reply:
If the yaml in question is the one from `penalty_details.yaml` you could do the following...
```
type PenaltyDetails struct {
	Name    string    `yaml:"name"`
	Details []Details `yaml:"details"`
}

type LP struct {
	Description        string `yaml:"Description"`
	DescriptionID      string `yaml:"DescriptionId"`
	ResourceKind       string `yaml:"ResourceKind"`
	ProductType        string `yaml:"ProductType"`
	EmailReceivedAppID string `yaml:"EmailReceivedAppId"`
	EmailFilingDesc    string `yaml:"EmailFilingDesc"`
	EmailMsgType       string `yaml:"EmailMsgType"`
}

type Details struct {
	LP LP `yaml:"LP"`
}
```

This would be used like...
```
	pd := models.PenaltyDetails{}
	...
	pd.Details[0].LP.Description = "This is a description"
	pd.Details[0].LP.DescriptionID = "6d623c778ed54dcaa96664939e734618"
	pd.Details[0].LP.ResourceKind = "some resource"

        ...
	for _, Detail := range pd.Details {
		fmt.Println(Detail.LP.Description)
	}

```

This would let you use

Reply:
if the object for details doesn't need to be an array/slice, you can just have `PenaltyDetails` have a `Details Details` so you can just go `pd.Detail.LP.Description`, however having Details as an array lets there be multiple details in the yaml

Reply:
So we are going to have at least another two entries in the YAML for 'CS' and 'ROE'. This is why I couldn't use the 'LP' struct as it variable if that makes sense.

Reply:
I have updated the map to use the struct as per our conversation.